docs/examples/module-temperature-check.py

```diff
@@ -0,0 +1,60 @@
+"""
+Module Temperature Check
+========================
+
+Test whether the module temperature is correlated with irradiance.
+"""
+
+# %%
+# Testing the correlation between module temperature and irradiance
+# measurements can help identify if there are issues with the module
+# temperature sensor.
+# In this example, we demonstrate how to use
+# :py:func:`pvanalytics.quality.weather.module_temperature_check`, which
+# runs a linear regression model of module temperature vs. irradiance. Model
+# performance is then assessed by correlation coefficient. If it meets a
+# minimum threshold, function outputs a True boolean. If not, it outputs a
+# False boolean.
+
+import pvanalytics
+from pvanalytics.quality.weather import module_temperature_check
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy.stats import linregress
+import pathlib
+
+# %%
+# First, we read in example data from the NREL SERF West system, which contains
+# data for module temperature and irradiance under the 'module_temp_1__781'
+# and 'poa_irradiance__771' columns, respectively. This data set contains
+# 15-minute averaged measurements, and is available via the NREL PVDAQ database
+# as system 51.
+
+pvanalytics_dir = pathlib.Path(pvanalytics.__file__).parent
+serf_east_file = pvanalytics_dir / 'data' / 'serf_west_15min.csv'
+data = pd.read_csv(serf_east_file, index_col=0, parse_dates=True)
+print(data[['module_temp_1__781', 'poa_irradiance__771']].head(10))
+
+# %%
+# We then use :py:func:`pvanalytics.quality.weather.module_temperature_check`
+# to regress module temperature against irradiance POA, and check if the
+# relationship meets the minimum correlation coefficient criteria.
+corr_coeff_bool = module_temperature_check(data['module_temp_1__781'],
+                                           data['poa_irradiance__771'])
+print("Passes correlation coeff threshold? " + str(corr_coeff_bool))
+
+# %%
+# We then plot module temperature against irradiance to illustrate the
+# relationship.
+data.plot(x='module_temp_1__781',
+          y='poa_irradiance__771',
+          style='o', legend=None)
+data_reg = data[['module_temp_1__781', 'poa_irradiance__771']].dropna()
+# Add the linear regression line
+reg = linregress(data_reg['module_temp_1__781'].values,
```

Comment:
I would print the correlation coefficient `ref.rvalue` into the figure.

It would also be good to clarify that the function examines the Pearson correlation coefficient, add to the docstring for `module_temperature_check`.

Reply:
Printed the Pearson coeff + added it to docstrings in https://github.com/pvlib/pvanalytics/pull/151/commits/553bbe8e28e8216cba86c8ae7be117e93d80aae4